src/rez/cli/pkg-cache.py

```diff
@@ -70,7 +78,15 @@ def add_variant(pkgcache, uri, opts):
         print("No such variant: %s" % uri, file=sys.stderr)
         sys.exit(1)
 
-    destpath, status = pkgcache.add_variant(variant, force=opts.force)
+    if opts.pkg_cache_mode is not None:
+        cache_mode = True if opts.pkg_cache_mode == "sync" else False
+        destpath, status = pkgcache.add_variant(
+            variant, force=opts.force,
+            wait_for_copying=cache_mode
+        )
+    # If no mode is specified, use the default behavior.
+    else:
+        destpath, status = pkgcache.add_variant(variant, force=opts.force)
```

Comment:
```suggestion
    if opts.pkg_cache_mode == "async":
        cache_mode = True
    elif opts.pkg_cache_mode == "sync":
        cache_mode = False
    else:
        cache_mode = not config.package_cache_async

    destpath, status = pkgcache.add_variant(
        variant, force=opts.force,
        wait_for_copying=cache_mode
    )
```
This would more properly do what we want. The code you wrote would default to the function's default which would not necessarily be `config.package_cache_async`.